pootle/apps/pootle_store/unit/timeline.py

```diff
@@ -45,173 +45,204 @@ class UnitTimelineLog(UnitLog):
 
-    def base_entry(self):
-        entry = self.entry_dict
-        entry['new_value'] = to_python(self.submission.new_value)
-        return entry
+        return dict(
+            value=self.submission.new_value,
+            translation=True)
 
-    def qc_entry(self):
-        entry = self.entry_dict
-        check_name = self.submission.qc_name
-        check_url = (
-            u''.join(
-                [reverse('pootle-checks-descriptions'),
-                 '#', check_name]))
-        entry.update(
-            {'check_name': check_name,
-             'check_display_name': CHECK_NAMES[check_name],
-             'checks_url': check_url})
-        return entry
 
-    def state_change_entry(self):
-        entry = self.entry_dict
-        entry['old_value'] = STATES_MAP[int(to_python(self.submission.old_value))]
-        entry['new_value'] = STATES_MAP[int(to_python(self.submission.new_value))]
-        return entry
+class UnitCreatedEntry(object):
+    def __init__(self, unit_source, **kwargs):
+        self.unit_source = unit_source
+        self.target_event = kwargs.get("target_event")
+
+    @property
+    def entry(self):
+        target_changed = (
+            self.target_event is not None
+            and self.target_event.value.old_value != '')
+        entry = dict(description=_("Unit created"))
+        if target_changed:
+            entry['value'] = self.target_event.value.old_value
+            entry['translation'] = True
+        else:
+            if self.unit_source.unit.istranslated():
+                entry['value'] = self.unit_source.unit.target
+                entry['translation'] = True
 
-    def suggestion_entry(self):
-        entry = self.entry_dict
-        entry.update({'suggestion_text': self.submission.suggestion_target})
         return entry
 
+
+class UnitStateChangedEntry(SubmissionEntry):
+
     @property
     def entry(self):
-        if self.submission.field == SubmissionFields.STATE:
-            return self.state_change_entry()
-        elif self.submission.suggestion:
-            return self.suggestion_entry()
-        elif self.submission.qc_name:
-            return self.qc_entry()
-        return self.base_entry()
+        params = {
+            'old_value': STATES_MAP[int(to_python(self.submission.old_value))],
+            'new_value': STATES_MAP[int(to_python(self.submission.new_value))]}
+        value = format_html(_(
+            u"%(old_value)s "
+            u"<span class='timeline-arrow'></span> "
+            u"%(new_value)s", params))
 
+        return dict(value=value,
+                    state=True)
 
-class Timeline(object):
 
-    entry_class = TimelineEntry
-    fields = SubmissionProxy.timeline_fields
+class CommentUpdatedEntry(SubmissionEntry):
+    @property
+    def entry(self):
+        if self.submission.new_value:
+            return dict(
+                value=self.submission.new_value,
+                sidetitle=_("Comment:"),
```

Comment:
Despite it might not be the best description, since if there is a previous comment you are actually replacing it.